File: TP1/palindromoRecursivo.java
import java.util.Scanner;

public class palindromoRecursivo {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String entrada = new String();

        entrada = sc.nextLine();

        while (true) {

            if (isFim(entrada)) {
                entrada = entrada.toLowerCase();
                entrada = entrada.replaceAll("-", "");
                break;
            }

            System.out.println(epalindromo(entrada, 0) ? "SIM" : "NAO"); 
            entrada = sc.nextLine();
        }
    }


    public static boolean epalindromo(String str, int i) {
        if (i < (str.length() / 2)) { 
            if (str.charAt(i) == str.charAt(str.length() - i - 1)) {
                return epalindromo(str, ++i);
            } else {
                return false; // ovo
            }
        }
        return true;
    }

    public static boolean isFim(String s) {
        return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }
}

File: TP1/palindromoRecursivo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

bool isFim(char *s)
{
    return (strlen(s) == 4 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}

bool epalindromo(char *s, int i)
{
    if (i < strlen(s) - 1)
    {
        if (s[i] == s[strlen(s) - 2 - i])
            return epalindromo(s, ++i);
        else
            return false;
    }
    return true;
    do
    {
        s[i] != s[strlen(s) - 2 - i];
        return true;
    } while (i < strlen(s) - 1 / 2, i++);
    return false;
}

int main()
{
    char entrada[1000][1000];
    int numEntrada = 0;

    do
    {
        fgets(entrada[numEntrada], 1000, stdin);
    } while (!isFim(entrada[numEntrada]));
    numEntrada--;

    for (int i = 0; i < numEntrada; i++)
    {
        printf("%s", epalindromo(entrada[i], 0) ? "SIM\n" : "NAO\n");
    }
    return 0;
}